packages/schema/src/Schema.ts

```diff
@@ -1870,8 +1870,27 @@ export const optional: {
     } | {
       readonly as: "Option"
       readonly default?: never
-      readonly exact?: true
-      readonly nullable?: true
+      readonly exact?: never
+      readonly nullable?: never
+      readonly onNoneEncoding?: undefined
+    } | {
+      readonly as: "Option"
+      readonly default?: never
+      readonly exact?: never
+      readonly nullable: true
+      readonly onNoneEncoding?: null | undefined
+    } | {
+      readonly as: "Option"
+      readonly default?: never
+      readonly exact: true
+      readonly nullable?: never
+      readonly onNoneEncoding?: never
+    } | {
+      readonly as: "Option"
+      readonly default?: never
+      readonly exact: true
+      readonly nullable: true
+      readonly onNoneEncoding?: null
```

Comment:
The types are more complex, mostly because of the exact case. I feel like it does not make sense to allow `onNoneEncoding` for the exact case when `nullable` is not true.
It could be challenged to add this option for the exact case too.